### PR TITLE
HW4 Testing

### PR DESCRIPTION
**Pull request by vm137**

### CODE REVIEW

**Review by loginov-rocks — CHANGES_REQUESTED** [review]
Please add `coverage` dir to `.gitignore` and remove them from this pull request.

File: package.json [hunk 1]
```diff
@@ -39,24 +41,38 @@
...
     "eslint-plugin-import": "^2.16.0",
     "eslint-plugin-jsx-a11y": "^6.2.1",
     "eslint-plugin-react": "^7.12.4",
     "file-loader": "^3.0.1",
     "html-webpack-plugin": "^3.2.0",
-    "jest": "^24.4.0",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^24.5.0",
     "node-sass": "^4.11.0",
+    "react-test-renderer": "^16.8.6",
```
Comment (loginov-rocks) [inline package.json]: `react-test-renderer` is not used, so it should be deleted from the dependencies list.

File: package.json [hunk 2]
```diff
@@ -39,24 +41,38 @@
...
     "sass-loader": "^7.1.0",
+    "sinon": "^7.3.1",
```
Comment (loginov-rocks) [inline package.json]: `sinon` is not used, so it should be deleted from the dependencies list.

File: package.json [hunk 3]
```diff
@@ -6,8 +6,10 @@
     "start": "webpack-dev-server --mode=development",
     "build": "webpack --mode=production",
     "lint": "eslint --ext js,jsx src",
-    "test": "jest",
-    "ci": "npm run lint && npm run build"
+    "test": "jest src --env=jsdom",
```
Comment (loginov-rocks) [inline package.json]: Jsdom env is used by Jest by default, so the arg is not neccessary

File: package.json [hunk 4]
```diff
@@ -6,8 +6,10 @@
...
+    "test:cover": "jest src --coverage",
+    "ci": "npm run lint && npm run build",
```
Comment (loginov-rocks) [inline package.json]: As we implemented tests we need to include them to our CI process, so please add the `npm run test` to the `ci` script.

**Comment by loginov-rocks** [comment]
I don't see Cypress tests for the app, only default tests that we don't need to include.

**Comment by vm137** [comment]
I've made all requested changes and added cypress test.

File: src/client/components/App.test.jsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { shallow, render } from 'enzyme';
+import React from 'react';
+import App from './App';
+
+describe('MovieCard', () => {
+  it('renders matching snapshot', () => {
+    const wrapper = render(<App />);
```
Comment (loginov-rocks) [inline src/client/components/App.test.jsx]: Please replace `render` to `shallow`

File: src/client/components/App.test.jsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { shallow, render } from 'enzyme';
+import React from 'react';
+import App from './App';
+
+describe('MovieCard', () => {
```
Comment (loginov-rocks) [inline src/client/components/App.test.jsx]: Please rename test to `App`

File: src/client/components/MovieCard/MovieCard.test.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import React from 'react';
+
+import MovieCard from './MovieCard';
+
+const onClick = () => {};
+const mv = {
+  id: 336, title: 'Duck, You Sucker', tagline: 'Rod Steiger and James Coburn will blow you apart in', vote_average: 7.7, vote_count: 250, release_date: '1971-10-20', poster_path: 'https://image.tmdb.org/t/p/w500/nQkr1yi87pX1d3lQVTrBEWlCpT1.jpg', overview: 'At the beginning of the 1913 Mexican Revolution, greedy bandit Juan Miranda and idealist John H. Mallory, an Irish Republican Army explosives expert on the lam from the British, fall in with a band of revolutionaries plotting to strike a national bank. When it turns out that the government has been using the bank as a hiding place for illegally detained political prisoners -- who are freed by the blast -- Miranda becomes a revolutionary hero against his will.', budget: 0, revenue: 0, genres: ['Western'], runtime: 157,
+};
+
+describe('MovieCard', () => {
```
Comment (loginov-rocks) [inline src/client/components/MovieCard/MovieCard.test.jsx]: Please test handleClick method.

File: src/client/components/SearchBlock/SearchBlock.test.jsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
+import { mount, render } from 'enzyme';
+import React from 'react';
+import SearchBlock from './SearchBlock';
+
+const movies = [];
+const numberFoundMovies = 10;
+const searchCB = () => {};
+const onClick = () => {};
+
+describe('SearchBlock', () => {
```
Comment (loginov-rocks) [inline src/client/components/SearchBlock/SearchBlock.test.jsx]: Please test state changes after `updateInputValue` triggering by user input.

File: src/client/components/SearchResults/SearchResults.test.jsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
+import { render } from 'enzyme/build';
+import React from 'react';
+import SearchResults from './SearchResults';
+
+const onClick = jest.fn();
+const movies = [
```
Comment (loginov-rocks) [inline src/client/components/SearchResults/SearchResults.test.jsx]: Please extract this to a separate file like `./__fixtures__/movies.json`

File: src/client/components/ErrorBoundary/ErrorBoundary.test.jsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+describe('<ErrorBoundary />', () => {
+  it('throws error', () => {
+    const error = new Error('test');
+    const wrapper = mount(
+      <ErrorBoundary>
+        <ChildComponent />
+      </ErrorBoundary>,
+    );
+    wrapper.find(ChildComponent).simulateError(error);
```
Comment (loginov-rocks) [inline src/client/components/ErrorBoundary/ErrorBoundary.test.jsx:15]: Nice job!

File: src/client/components/SearchBlock/SearchBlock.test.jsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+import SearchBlock from './SearchBlock';
+
+const movies = [];
+const numberFoundMovies = 10;
+const searchCB = () => {};
+const onClick = () => {};
+
+describe('<SearchBlock />', () => {
+  it('renders matching snapshot', () => {
+    const wrapper = render(<SearchBlock
```
Comment (loginov-rocks) [inline src/client/components/SearchBlock/SearchBlock.test.jsx]: Still using `render`

File: src/client/components/SearchResults/SearchResults.test.jsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+  it('renders matching snapshot', () => {
+    const wrapper = shallow(<SearchResults
+      movies={testMovies}
+      onClick={onClick}
+    />);
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('check for .movie-card-wrapper', () => {
+    const wrapper = render(<SearchResults
```
Comment (loginov-rocks) [inline src/client/components/SearchResults/SearchResults.test.jsx:18]: Still using `render`

File: src/client/components/SingleMovie/SingleMovie.test.jsx [hunk 13]
```diff
@@ -0,0 +1,47 @@
+import { shallow, render } from 'enzyme';
+import React from 'react';
+import SingleMovie from './SingleMovie';
+
+const movieId = 336;
+const onClick = () => {};
+
+describe('<SingleMovie />', () => {
+  it('renders matching snapshot', () => {
+    const wrapper = render(<SingleMovie
```
Comment (loginov-rocks) [inline src/client/components/SingleMovie/SingleMovie.test.jsx]: Still using `render`

**Review by loginov-rocks — CHANGES_REQUESTED** [review]
It seems `shallow()` doesn't return snapshot to compare, we have to add any serializer, I suggest https://www.npmjs.com/package/enzyme-to-json - please install it as dev dependency and add the following to package.json -> jest prop:
```
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
```
After that you can update snapshots and they will content JSON tree instead of `ShallowWrapper {}`

File: src/client/components/ErrorBoundary/ErrorBoundary.test.jsx (hunk 10 above)
Comment (vm137) [reply src/client/components/ErrorBoundary/ErrorBoundary.test.jsx:15]: cool! thank you.

File: src/client/components/SearchBlock/SearchBlock.test.jsx (hunk 11 above)
Comment (vm137) [reply src/client/components/SearchBlock/SearchBlock.test.jsx]: changed for mount()

File: src/client/components/SingleMovie/SingleMovie.test.jsx (hunk 13 above)
Comment (vm137) [reply src/client/components/SingleMovie/SingleMovie.test.jsx]: changed for shallow()

**Review by loginov-rocks — APPROVED** [review]
Congratulations!

**Pull request merged**